### PR TITLE
Pre-released - Full CI/CD suite validation and smoke testing for next release

### PR DESCRIPTION
**Pull request by sb-gaurava**
# bugsnag-js `next` Branch — Cumulative PR Review
> **Repo:** [bugsnag/bugsnag-js](https://github.com/bugsnag/bugsnag-js)
> **Branch:** `next`
> **Review Date:** 2026-08-20
> **Total PRs merged since last release:** 8
> **PRs reviewed:** [#2808](#pr-2808), [#2790](#pr-2790), [#2787](#pr-2787), [#2784](#pr-2784), [#2773](#pr-2773), [#2769](#pr-2769), [#2752](#pr-2752), [#2747](#pr-2747)

---

## Table of Contents
1. [PR Summary Table](#pr-summary-table)
2. [Detailed PR Breakdown](#detailed-pr-breakdown)
3. [Cumulative File Changes](#cumulative-file-changes)
4. [Correctness Assessment](#correctness-assessment)
5. [Validation Steps](#validation-steps)
6. [Testing Steps](#testing-steps)
7. [Known Follow-ups & Open Items](#known-follow-ups--open-items)

---

## PR Summary Table

| # | PR | Title | Author | Merged | Category |
|---|-----|-------|--------|--------|----------|
| 1 | [#2808](https://github.com/bugsnag/bugsnag-js/pull/2808) | PLAT-17038 — Tight Retry Loop / Exponential Backoff on Minidump Delivery | sb-gaurava | Aug 19, 2026 | Bug Fix / Reliability |
| 2 | [#2790](https://github.com/bugsnag/bugsnag-js/pull/2790) | iOS React Native Navigation Fixes | SB-priyankap | Jul 3, 2026 | CI / RN Fix |
| 3 | [#2787](https://github.com/bugsnag/bugsnag-js/pull/2787) | Fix SCA Findings — React Native Version Update | SB-priyankap | Jul 17, 2026 | SCA / Security |
| 4 | [#2784](https://github.com/bugsnag/bugsnag-js/pull/2784) | Update `form-data` for SCA Finding | SB-priyankap | Jun 25, 2026 | SCA / Security |
| 5 | [#2773](https://github.com/bugsnag/bugsnag-js/pull/2773) | Use `bugsnag-cli` Instead of Legacy `bugsnag-source-maps` | SB-gulshans | Jul 3, 2026 | Tooling Migration |
| 6 | [#2769](https://github.com/bugsnag/bugsnag-js/pull/2769) | fix(plugin-electron-client-state-persistence): Unbreak Build on glibc ≥ 2.28 | SB-priyankap | Jul 13, 2026 | Bug Fix / Linux Build |
| 7 | [#2752](https://github.com/bugsnag/bugsnag-js/pull/2752) | PLAT-16334 — Mazerunner Version Update to 11.2 | SB-khushbum | May 25, 2026 | CI / Test Infra |
| 8 | [#2747](https://github.com/bugsnag/bugsnag-js/pull/2747) | Replace Deprecated `android_9` for BrowserStack | SB-khushbum | May 8, 2026 | CI / Test Infra |

---

## Detailed PR Breakdown

---

### PR #2808
**Title:** PLAT-17038-MinidumpDeliveryLoop — Tight Retry Loop on Unreachable Endpoint
**Author:** sb-gaurava | **Merged:** Aug 19, 2026
**Link:** https://github.com/bugsnag/bugsnag-js/pull/2808

#### Why this change was made
When the Electron minidump delivery endpoint was unreachable or returning `5xx` errors, the delivery loop was **tight-retrying with no delay** — hammering the server, spiking CPU to ~87–119%, and flooding logs with ~57 failed attempts/sec. This needed exponential backoff with full jitter to prevent thundering herd.

#### ✅ What was ADDED

**`packages/plugin-electron-deliver-minidumps/minidump-loop.js`**
- Constants: `BACKOFF_BASE_MS = 1000`, `BACKOFF_MAX_MS = 60000`, `BACKOFF_FACTOR = 2`
- `calculateBackoff(failureCount)` — returns `random value in [0, min(60s, 1s × 2ⁿ)]` with full jitter
- Instance state: `this._failureCount` and `this._nextDelay`
- `_scheduleSelf()` call to throttle loop iterations
- On `_onerror`: increment `_failureCount`, compute `_nextDelay`, log retry info
- On non-retryable errors: reset both `_failureCount` and `_nextDelay` to `0`
- On `start()`: reset `_failureCount` and `_nextDelay` to `0` (fresh slate)

**`test/minidump-loop.test.ts`**
- `flushAll()` helper (10× `flushPromises`)
- `stepLoop()` / `advancePastBackoff()` helpers
- `createPersistentQueue(minidump)` helper
- `peek()` stubbed to wait until `remove()` is called (needed for retry scenarios)
- `logger.info` spy on test logger
- New `describe('exponential backoff behaviour')` test suite

**`package.json` (root)**
- Upgraded `lerna` from `^8.1.8` → `^10.0.0`
- Fixed `react-native` override to use `$variable` references correctly
- Added overrides: `brace-expansion: 2.0.2`, `js-yaml: 4.3.1` under `eslint` scope, `tar: ^7.5.16`, `minimatch@>=10: { brace-expansion: ^5.0.8 }`

#### 🔄 What was UPDATED

**`packages/plugin-electron-deliver-minidumps/minidump-loop.js`**
- `_scheduleSelf()` now accepts and uses a `delay` parameter
- `setTimeout` is now the actual throttle mechanism (was immediate re-schedule before)

**`test/minidump-loop.test.ts`**
- `'attempts redelivery'` test updated to use `advancePastBackoff()` to account for new delay between retries

**`package-lock.json`**
- Rebuilt on `node:22-alpine` (npm 10.9.7)
- Dropped `--legacy-peer-deps` — `npm ci --unsafe-perm` now succeeds cleanly

#### 🗑️ What was REMOVED and Why

**`package.json` (root) — stale overrides removed:**
- `"brace-expansion": "^2.1.2"` — replaced by exact pin `2.0.2`
- `"js-yaml": "^4.3.0"` (standalone) — scoped to `eslint` only where needed

**`package-lock.json` — stale entries removed:**
- `@angular-devkit/build-webpack`, `jsonc-parser`, `ora`, `sass@1.85.0`, `esbuild@0.28.0`, nested `@esbuild/*` platform packages, `yoctocolors-cjs`
- `"hasInstallScript": true` from electron plugin packages — `node-gyp` bindings not needed in lockfile metadata

**Why removed:** These were stale/incorrect lockfile artifacts from previous dependency resolutions that caused `npm ci` to fail on clean `node:22-alpine` environments.

#### Bundle Size Impact
| | Minified | Minified + Gzipped |
|--|----------|-------------------|
| Before | 51.46 kB | 15.30 kB |
| After | 51.46 kB | 15.30 kB |
| Delta | No change | No change |

#### Code Coverage Impact
| File | Lines | Branches | Functions | Statements |
|------|-------|----------|-----------|------------|
| `minidump-loop.js` | 95.83% (+4.17%) | 85.18% (+3.7%) | 100% | 95.94% (+3.94%) |

---

### PR #2790
**Title:** iOS React Native Navigation Fixes
**Author:** SB-priyankap | **Merged:** Jul 3, 2026
**Link:** https://github.com/bugsnag/bugsnag-js/pull/2790

#### Why this change was made
The React Native Navigation iOS old architecture fixture builds and E2E tests were failing in CI due to incompatibility with updated React Native versions. Rather than blocking the pipeline, the steps were removed while a new-arch replacement is investigated.

#### ✅ What was ADDED
- `CHANGELOG.md`: New `[Unreleased] Fixed` entry for React Native sourcemap upload fix

#### 🗑️ What was REMOVED and Why

**`.buildkite/basic/react-native-ios-full-pipeline.yml`**
- Removed: `react-native-navigation` iOS old architecture fixture build step
- Removed: `react-native-navigation` iOS old architecture E2E test step

**Why removed:** These steps were failing due to build incompatibilities when using `react-native-navigation` with the old architecture on the updated RN version. The new-arch equivalent blocks remain commented out with a `TODO` to re-enable when RN test support is extended.

> ⚠️ **Note:** This leaves no active iOS E2E coverage for `react-native-navigation`. A follow-up is needed to add a replacement E2E job against a supported RN/RNN combination.

---

### PR #2787
**Title:** Fix SCA Findings — React Native Version Update
**Author:** SB-priyankap | **Merged:** Jul 17, 2026
**Link:** https://github.com/bugsnag/bugsnag-js/pull/2787

#### Why this change was made
SCA (Software Composition Analysis) scan flagged several high-priority vulnerabilities in transitive dependencies of React Native. This PR updates dependencies and adjusts tooling (TypeScript, Babel, ESLint, Jest) to keep the monorepo building and CI passing post-upgrade.

#### ✅ What was ADDED
- `tsconfig.eslint.json` — broad include patterns for type-aware linting across `packages/` and root
- `test/browser/features/fixtures/http_errors/package.json` — override pins for vulnerable transitive deps in test fixture
- `packages/electron-test-helpers/src/BrowserWindow.ts` — `_isDestroyed` internal state to the fake `BrowserWindow`
- `.github/workflows/test-electron.yml` — added explicit Electron install step
- `.buildkite/basic/electron-pipeline.yml` — added explicit Electron install step

#### 🔄 What was UPDATED and Why

| File | What Changed | Why |
|------|-------------|-----|
| `package.json` | Updated `react-native`, `typescript`, Babel plugins; added SCA-pin overrides | Address SCA findings, keep CI green |
| `tsconfig.json` | Adjusted include/exclude patterns | Support monorepo-wide typecheck after RN update |
| `.eslintrc.js` | Reworked ignore patterns, scoped Jest rules to test files only, restored promise/unsafe rules at `warn` level | Reduce lint friction post-RN update; promise rules at `warn` to surface issues without blocking CI |
| `babel.config.js` | Replaced `proposal-*` plugins with `transform-*` plugins; added RN-specific overrides for tests | Babel no longer ships proposal plugins separately in newer versions |
| `jest.config.js` | Added RN transform ignore pattern | Support updated RN/Jest transforms |
| `lerna.json` | Added `--legacy-peer-deps` to publish args | Avoid peer dep conflicts during publish |
| `packages/delivery-electron/test/queue.test.ts` | Refactored async handling, updated Jest matchers | Jest API alignment post-upgrade |
| All test files (20+) | `toBeCalledWith` → `toHaveBeenCalledWith`, `toBeCalledTimes` → `toHaveBeenCalledTimes` | Jest deprecated the old aliases |

#### 🗑️ What was REMOVED and Why
- `--report-unused-disable-directives` temporarily removed then **restored** after review feedback — stale `eslint-disable` comments were cleaned up instead (50 removed, 1 targeted suppression kept)
- `.npmrc` `legacy-peer-deps=true` was added then **removed** after review — RN-related dependency pins were loosened to caret ranges instead, giving cleaner install behavior
- `strict: true` / `strictNullChecks: true` / `noImplicitAny: true` in `tsconfig.json` — **temporarily relaxed** to unblock CI during upgrade

> ⚠️ **Temporary relaxation tracked in:** [PLAT-16909](https://smartbear.atlassian.net/browse/PLAT-16909) — restore `strictNullChecks` → `noImplicitAny` → `strict` incrementally as type debt is reduced.

#### Bundle Size Impact
| | Minified | Minified + Gzipped |
|--|----------|-------------------|
| Before | 51.76 kB | 15.31 kB |
| After | 51.46 kB | 15.30 kB |
| Delta | **-299 bytes** | **-1 byte** |

---

### PR #2784
**Title:** Update `form-data` for SCA Finding
**Author:** SB-priyankap | **Merged:** Jun 25, 2026
**Link:** https://github.com/bugsnag/bugsnag-js/pull/2784

#### Why this change was made
SCA scan flagged a vulnerability in the `form-data` transitive dependency. The version was pinned/updated in `package.json` to remediate the finding.

#### 🔄 What was UPDATED
- `package.json` — `form-data` version updated/pinned to a non-vulnerable version
- `package-lock.json` — updated to reflect new resolution

#### Why this matters
`form-data` is used in delivery payloads. A vulnerable version could expose multipart form data handling to known CVEs.

---

### PR #2773
**Title:** Use `bugsnag-cli` Instead of Legacy `bugsnag-source-maps`
**Author:** SB-gulshans | **Merged:** Jul 3, 2026
**Link:** https://github.com/bugsnag/bugsnag-js/pull/2773

#### Why this change was made
The legacy `bugsnag-source-maps` tool is deprecated. The React Native iOS Xcode build-phase upload script was still using the old tool and old flag names, causing upload failures. Migrated to `bugsnag-cli` which is the actively maintained replacement.

#### 🔄 What was UPDATED

**React Native iOS Xcode build-phase upload script**
- Upload command changed: `bugsnag-source-maps upload` → `bugsnag-cli upload react-native-ios`
- Flag names updated to match the Bugsnag CLI interface:
  - Bundle/app version naming flags updated
  - Upload URL flag updated

#### 🗑️ What was REMOVED and Why
- References to `bugsnag-source-maps` — deprecated tool, no longer maintained, incompatible with current upload API

---

### PR #2769
**Title:** fix(plugin-electron-client-state-persistence): Unbreak Build on glibc ≥ 2.28
**Author:** SB-priyankap | **Merged:** Jul 13, 2026
**Link:** https://github.com/bugsnag/bugsnag-js/pull/2769

#### Why this change was made
`@bugsnag/plugin-electron-client-state-persistence` failed to compile on any system with **glibc ≥ 2.28** (Ubuntu 18.10+, Debian 10+, Fedora 29+, Arch Linux, etc.).

**Root cause:** The vendored `tinycthread` library redefined `once_flag` and `call_once` on POSIX — names that glibc now exposes natively through `<stdlib.h>`, producing `conflicting types for 'pthread_once_t'` errors during `node-gyp rebuild`. This broke `npm install` for any consumer of `@bugsnag/electron` on a current Linux distro.

#### 🗑️ What was REMOVED and Why

**`packages/plugin-electron-client-state-persistence/src/deps/tinycthread/tinycthread.h`**
- Removed the two POSIX `#else` branches (5 lines total) that redefine `once_flag` / `ONCE_FLAG_INIT` / `call_once`

**Why safe to remove:** The plugin only uses `mtx_t`, `mtx_init`, `mtx_lock`, `mtx_unlock`, and `mtx_plain` from `tinycthread`. The `once_flag`/`call_once` definitions were **dead code on POSIX** — the only active `call_once` definition in `tinycthread.c` is gated behind `#if defined(TTHREAD_WIN32)`. Win32 path is **untouched**.

---

### PR #2752
**Title:** PLAT-16334 — Mazerunner Version Update to 11.2
**Author:** SB-khushbum | **Merged:** May 25, 2026
**Link:** https://github.com/bugsnag/bugsnag-js/pull/2752

#### Why this change was made
Mazerunner (BugSnag's E2E test framework) was on an outdated version. Upgrading to 11.2 brings in test runner improvements and aligns with the current supported version used across other BugSnag repos.

#### 🔄 What was UPDATED
- Mazerunner version pinned to `11.2` in relevant CI configuration files
- `browser_errors.yml` updated: Android device version changed from `android_9` → `android_13`

---

### PR #2747
**Title:** Replace Deprecated `android_9` for BrowserStack
**Author:** SB-khushbum | **Merged:** May 8, 2026
**Link:** https://github.com/bugsnag/bugsnag-js/pull/2747

#### Why this change was made
BrowserStack deprecated the `android_9` device profile. CI tests using it were failing or would fail imminently. Replaced with `android_13` which is actively supported.

#### 🔄 What was UPDATED
- All BrowserStack device mappings: `android_9` → `android_13`
- `TESTING.md` — updated test command examples to reference `android_13` and `ANDROID_16`

#### 🗑️ What was REMOVED and Why
- All references to `android_9` device profile — deprecated by BrowserStack, no longer available for test runs

---

### Core Package Files
| File | PRs | Change Type |
|------|-----|-------------|
| `package.json` (root) | #2808, #2787, #2784 | Updated — dependency versions, overrides, lerna upgrade |
| `package-lock.json` | #2808, #2787, #2784 | Updated — rebuilt on node:22-alpine, stale entries removed |
| `lerna.json` | #2787 | Updated — added `--legacy-peer-deps` to publish args |
| `tsconfig.json` | #2787 | Updated — include/exclude patterns, strict temporarily relaxed |
| `tsconfig.eslint.json` | #2787 | Added — type-aware linting support |
| `.eslintrc.js` | #2787 | Updated — scoped Jest rules, restored promise rules at warn |
| `babel.config.js` | #2787 | Updated — proposal → transform plugins |
| `jest.config.js` | #2787 | Updated — RN transform ignore pattern |
| `CHANGELOG.md` | #2790 | Updated — Unreleased Fixed entry |

### Plugin/Package Files
| File | PRs | Change Type |
|------|-----|-------------|
| `packages/plugin-electron-deliver-minidumps/minidump-loop.js` | #2808 | Updated — exponential backoff added |
| `packages/plugin-electron-deliver-minidumps/test/minidump-loop.test.ts` | #2808, #2787 | Updated — backoff tests + Jest matcher fixes |
| `packages/plugin-electron-deliver-minidumps/test/deliver-minidumps.test.ts` | #2787 | Updated — Jest matchers |
| `packages/plugin-electron-deliver-minidumps/test/send-minidump.test.ts` | #2787 | Updated — Jest matchers |
| `packages/plugin-electron-client-state-persistence/src/deps/tinycthread/tinycthread.h` | #2769 | Updated — removed dead POSIX redefinitions |
| `packages/plugin-electron-ipc/test/bugsnag-ipc-main.test.ts` | #2787 | Updated — Jest matchers |
| `packages/plugin-electron-session/test/session.test.ts` | #2787 | Updated — Jest matchers |
| `packages/plugin-electron-net-breadcrumbs/test/net-breadcrumbs.test-main.ts` | #2787 | Updated — test descriptions |
| `packages/plugin-electron-network-status/test/network-status.test.ts` | #2787 | Updated — type spacing |
| `packages/delivery-electron/test/queue.test.ts` | #2787 | Updated — async handling + matcher updates, some cases removed |
| `packages/delivery-electron/queue.js` | #2787 | Coverage decreased (-1.82% lines) |
| `packages/electron-test-helpers/src/BrowserWindow.ts` | #2787 | Updated — added `_isDestroyed` state |
| `packages/electron-test-helpers/src/client.ts` | #2787 | Updated — returns client from session delegate methods |
| `packages/plugin-angular/package.json` | #2787 | Updated — Angular 19 toolchain deps |
| `packages/plugin-react/src/test/index.test.tsx` | #2787 | Updated — Jest matchers |
| `packages/plugin-react-navigation/test/react-navigation.test.tsx` | #2787 | Updated — Jest matchers |
| `packages/react-native-cli/src/lib/test/Xcode.test.ts` | #2787 | Updated — Jest matchers |
| `packages/react-native-cli/src/lib/test/Npm.test.ts` | #2787 | Updated — Jest matchers |
| `packages/web-worker/types/notifier.ts` | #2787 | Updated — type-only export |

### CI / Pipeline Files
| File | PRs | Change Type |
|------|-----|-------------|
| `.buildkite/basic/react-native-ios-full-pipeline.yml` | #2790 | Updated — removed old-arch RNN build + E2E steps |
| `.buildkite/basic/electron-pipeline.yml` | #2787 | Updated — added Electron install step |
| `.github/workflows/test-electron.yml` | #2787 | Updated — reordered install steps, added Electron install |
| `.github/workflows/pr-diff.yml` | #2787 | Updated — cleanup to avoid Electron file locks, reduced test worker parallelism |
| BrowserStack device configs | #2747, #2752 | Updated — android_9 → android_13, mazerunner 11.2 |
| `TESTING.md` | #2747 | Updated — device command examples |
| `browser_errors.yml` | #2752 | Updated — Android version 9 → 13 |

### RN iOS Source Map Upload Script
| File | PRs | Change Type |
|------|-----|-------------|
| RN iOS Xcode build-phase upload script | #2773 | Updated — `bugsnag-source-maps` → `bugsnag-cli` |

---

## Correctness Assessment

| PR | Assessment | Notes |
|----|-----------|-------|
| #2808 | ✅ **Correct** | Backoff math verified with pinned `Math.random`. Coverage improved. No bundle size regression. |
| #2790 | ✅ **Correct** | Correct to remove failing old-arch steps. ⚠️ Follow-up needed for RNN iOS E2E coverage gap. |
| #2787 | ⚠️ **Conditionally Correct** | SCA remediation goal met. TypeScript strict relaxation is a **temporary** regression tracked in [PLAT-16909](https://smartbear.atlassian.net/browse/PLAT-16909). `delivery-electron/queue.js` coverage dropped (-1.82% lines) — some test cases removed. |
| #2784 | ✅ **Correct** | Targeted SCA pin. Low risk. |
| #2773 | ✅ **Correct** | Migrating to the actively maintained `bugsnag-cli` is the right long-term move. Flag renames are correct per CLI docs. |
| #2769 | ✅ **Correct** | Dead-code removal only. Win32 path untouched. Verified on Arch Linux (glibc 2.43). |
| #2752 | ✅ **Correct** | Routine test infra upgrade. No production code changed. |
| #2747 | ✅ **Correct** | BrowserStack deprecated `android_9`; replacement with `android_13` is correct. |

---

### 1. Install & Build
```bash
# Clean install on node:22-alpine
npm ci --unsafe-perm

# Build all packages
npm run build

# Build Electron-specific packages
npm run build:electron
```

### 2. Lint
```bash
# Must pass with --report-unused-disable-directives restored
npm run test:lint
# Expect: 0 errors, 0 warnings (or warn-level only for restored promise/unsafe rules)
```

### 3. Type Check
```bash
npx tsc --noEmit
# Note: strict/strictNullChecks temporarily relaxed — track via PLAT-16909
```

### 4. Unit Tests
```bash
npm test
# Verify:
# - minidump-loop.test.ts: all backoff tests pass
# - deliver-minidumps.test.ts: all Jest matcher updates pass
# - queue.test.ts: async tests pass
# - All 20+ updated test files pass
```

### 5. Electron Build Validation
```bash
cd packages/plugin-electron-client-state-persistence
npm install
npx node-gyp rebuild
# Expected: produces build/Release/bugsnag_pecsp_bindings.node
# Test on: Linux (glibc ≥ 2.28), macOS, Windows
```

### 6. Minidump Delivery Backoff Validation
```bash
# Run backoff unit tests specifically
npx jest packages/plugin-electron-deliver-minidumps/test/minidump-loop.test.ts --verbose
# Verify:
# - calculateBackoff(0) returns value in [0, 1000]
# - calculateBackoff(6) returns value in [0, 60000]
# - _failureCount resets to 0 on start()
# - _failureCount resets to 0 on non-retryable error
```

### 7. Source Map Upload Validation
```bash
# Verify bugsnag-cli is installed and accessible
bugsnag-cli --version

# Test RN iOS upload script with dry-run
bugsnag-cli upload react-native-ios --help
# Verify flag names match updated script
```

### 8. BrowserStack Device Availability
```bash
# Confirm android_13 device is available
# Run: DEVICE=android_13 npm run test:browser
# Run: DEVICE=ANDROID_16 npm run test:browser
```

### 9. CI Pipeline Check
- Verify GitHub Actions `test-electron.yml` completes on `next` branch push
- Verify Buildkite `electron-pipeline.yml` completes successfully
- Verify `pr-diff.yml` runs without Electron file lock errors
- Verify Mazerunner 11.2 E2E tests run on `android_13`

---

### Unit Tests — Full Suite
```bash
npm test
```
Expected: All tests pass. Check specifically:
- `packages/plugin-electron-deliver-minidumps` — backoff + delivery tests
- `packages/delivery-electron` — queue tests (note: some cases removed in #2787)
- `packages/plugin-react` — Jest matcher updates
- `packages/plugin-angular` — Angular 19 toolchain

### Integration Tests — Electron
```bash
npm run test:electron
```
Expected: Electron install step runs first, then all electron integration tests pass.

### E2E Tests — BrowserStack Android
```bash
DEVICE=android_13 npm run test:browser:e2e
```
Expected: All E2E browser tests pass on `android_13`.

### Manual Backoff Verification
1. Start Electron app in dev mode
2. Point delivery endpoint to an unreachable host (e.g., `localhost:9999`)
3. Trigger a minidump
4. Observe logs — should see retry attempts with increasing delays: `1s → 2s → 4s → … → 60s`
5. CPU usage should remain low (< 5%) during retry loop

### Manual Source Map Upload Verification
1. Build a React Native iOS app with the updated Xcode build phase script
2. Trigger an archive build
3. Confirm `bugsnag-cli upload react-native-ios` runs without errors
4. Confirm source maps appear in the BugSnag dashboard

### Linux Build Verification (glibc ≥ 2.28)
```bash
# On Ubuntu 20.04+ / Arch Linux / Fedora 30+
cd packages/plugin-electron-client-state-persistence
npm install
npx node-gyp rebuild
# Expected: SUCCESS — build/Release/bugsnag_pecsp_bindings.node created
# Previous: FAIL — conflicting types for 'pthread_once_t'
```

---

## Known Follow-ups & Open Items

| # | Item | Ticket | Priority |
|---|------|--------|----------|
| 1 | Restore TypeScript `strict` / `strictNullChecks` / `noImplicitAny` | [PLAT-16909](https://smartbear.atlassian.net/browse/PLAT-16909) | High |
| 2 | Re-enable `ban-ts-comment` ESLint rule; migrate `@ts-ignore` → `@ts-expect-error` | [PLAT-16909](https://smartbear.atlassian.net/browse/PLAT-16909) | Medium |
| 3 | Add replacement iOS E2E coverage for `react-native-navigation` (new arch) | TBD | Medium |
| 4 | Restore removed `queue.test.ts` test cases (`isValidPayload` edge cases) | TBD | Medium |
| 5 | Consider replacing vendored `tinycthread` with system `<threads.h>` on C11 platforms | TBD | Low |
| 6 | Validate `bugsnag-cli` source map upload in production RN iOS release build | TBD | High |

---

[PLAT-16909]: https://smartbear.atlassian.net/browse/PLAT-16909?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: scripts/react-native/ios-utils.js [hunk 1]
```diff
@@ -42,57 +47,71 @@ module.exports = {
...
-      const applyPatch = ['apply', '--ignore-whitespace', resolve(replacementFilesDir, 'patches/react-native-boost.patch')]
-      execFileSync('git', applyPatch, { cwd: fixtureDir, stdio: 'inherit' })
-
+ 
+      // fix boost issues - update the boost download URL directly
+      const boostPodspecPath = resolve(fixtureDir, 'node_modules/react-native/third-party-podspecs/boost.podspec')
+      if (fs.existsSync(boostPodspecPath)) {
+        let boostPodspecContents = fs.readFileSync(boostPodspecPath, 'utf8')
+        boostPodspecContents = boostPodspecContents.replace(
+          'https://boostorg.jfrog.io/artifactory/main/release/1.76.0/source/boost_1_76_0.tar.bz2',
```
Comment (SB-jigneshR) [inline scripts/react-native/ios-utils.js:89]: @sb-gaurava / @SB-priyankap : why we need to provide specific version?

**Review by SB-jigneshR — COMMENTED** [review]
Please check comment

File: scripts/react-native/ios-utils.js (hunk 1 above)
Comment (sb-gaurava) [reply scripts/react-native/ios-utils.js:89]: Hi @SB-jigneshR,

The specific Boost version (`1.76.0`) is hardcoded because it matches the exact version and checksum expected by React Native's internal `boost.podspec` / `boost-for-react-native.podspec` for this RN version range (used by Folly / React Native core C++ dependencies).

**Why the URL replacement is necessary:**
The default upstream URL in older React Native podspecs points to `boostorg.jfrog.io`, which frequently experiences outages, rate-limiting, and 403/404 errors during CI runs. 

We patch the podspec in `ios-utils.js` to point to a reliable, stable mirror (such as SourceForge / AWS S3 archive) while keeping the exact version (`1.76.0`) and checksum required by React Native so `pod install` doesn't fail.

**Comment by sb-gaurava** [comment]
Raised the Release PR - https://github.com/bugsnag/bugsnag-js/pull/2822.

Closing this PR
